compiler/rustc_next_trait_solver/src/solve/assembly/structural_traits.rs

```diff
@@ -135,13 +138,20 @@ where
         | ty::Dynamic(_, _, ty::DynStar)
         | ty::Error(_) => Ok(ty::Binder::dummy(vec![])),
 
-        ty::Str
-        | ty::Slice(_)
-        | ty::Dynamic(..)
-        | ty::Foreign(..)
-        | ty::Alias(..)
-        | ty::Param(_)
-        | ty::Placeholder(..) => Err(NoSolution),
+        // impl {Meta,}Sized for str, [T], dyn Trait
+        ty::Str | ty::Slice(_) | ty::Dynamic(..) => match sizedness {
+            SizedTraitKind::Sized => Err(NoSolution),
+            SizedTraitKind::MetaSized => Ok(ty::Binder::dummy(vec![])),
+            SizedTraitKind::PointeeSized => unreachable!(),
```

Comment:
when do we want the `PointeeSized` variant of this enum? I feel like pretty much every use site shouldn't consider handling it

Reply:
Removed that variant of `SizedTraitKind` and removed a bunch of `unreachable` branches.